### PR TITLE
feat(FN-980): update error handling

### PR DESCRIPTION
**Pull request by francescastocco**
## Introduction
Improve error handling of previous reports page

## Resolution
- Not return 200 if something has gone wrong

### CODE REVIEW

File: dtfs-central-api/src/v1/controllers/portal/utilisation-report-service/previous-reports.controller.js [hunk 1]
```diff
@@ -17,11 +17,10 @@ const getUtilisationReports = async (req, res) => {
         },
       },
     ]).toArray();
-    return res.status(200).send(filteredAndSortedUtilisationReports);
+    res.status(200).send(filteredAndSortedUtilisationReports);
   } catch (error) {
     console.error('Unable to get utilisation reports %s', error);
```
Comment (AlexBramhill) [inline dtfs-central-api/src/v1/controllers/portal/utilisation-report-service/previous-reports.controller.js:22]: Any reason we're not returning a 500 here?

File: portal-api/src/v1/controllers/utilisation-report-service/previous-reports.controller.js [hunk 2]
```diff
@@ -55,7 +55,6 @@ const getPreviousReportsByBankId = async (req, res) => {
   } catch (error) {
     console.error('Unable to get previous reports %s', error);
```
Comment (AlexBramhill) [inline portal-api/src/v1/controllers/utilisation-report-service/previous-reports.controller.js:56]: Any reason for not returning a 500 here?

File: dtfs-central-api/src/v1/controllers/portal/utilisation-report-service/previous-reports.controller.js [hunk 3]
```diff
@@ -17,11 +17,11 @@ const getUtilisationReports = async (req, res) => {
...
+    res.status(500).send({ status: 500, message: error });
```
Comment (ChristianMcc) [inline dtfs-central-api/src/v1/controllers/portal/utilisation-report-service/previous-reports.controller.js:23]: As mentioned in person - can we change returning the code error to a generic 'Failed to get utilisation reports' style string

File: portal-api/src/v1/controllers/utilisation-report-service/previous-reports.controller.js [hunk 4]
```diff
@@ -54,8 +54,8 @@ const getPreviousReportsByBankId = async (req, res) => {
     res.status(200).send(reportsGroupedByYear.sort((a, b) => b.year - a.year));
   } catch (error) {
     console.error('Unable to get previous reports %s', error);
+    res.status(500).send({ status: 500, message: error });
```
Comment (ChristianMcc) [inline portal-api/src/v1/controllers/utilisation-report-service/previous-reports.controller.js:57]: ditto

**Review by ChristianMcc — CHANGES_REQUESTED** [review]
When we return an error to the frontend are we display as if there are no reports or displaying an error page? I'd semi expect that if an API call fails and something goes wrong that we would display the 'something went wrong' page rather than say 0 reports

**Comment by francescastocco** [comment]
> When we return an error to the frontend are we display as if there are no reports or displaying an error page? I'd semi expect that if an API call fails and something goes wrong that we would display the 'something went wrong' page rather than say 0 reports

I've moved the api call into the try/catch so will catch any errors and render the `problem-with-service` page

File: dtfs-central-api/src/v1/controllers/portal/utilisation-report-service/previous-reports.controller.js [hunk 5]
```diff
@@ -17,11 +17,11 @@ const getUtilisationReports = async (req, res) => {
         },
       },
     ]).toArray();
-    return res.status(200).send(filteredAndSortedUtilisationReports);
+    res.status(200).send(filteredAndSortedUtilisationReports);
   } catch (error) {
-    console.error('Unable to get utilisation reports %s', error);
+    console.error('Unable to get utilisation reports %O', error);
```
Comment (abhi-markan) [inline dtfs-central-api/src/v1/controllers/portal/utilisation-report-service/previous-reports.controller.js:22]: Looks like we are using `%s` everywhere else.

File: portal-api/src/v1/controllers/utilisation-report-service/previous-reports.controller.js [hunk 6]
```diff
@@ -53,9 +53,9 @@ const getPreviousReportsByBankId = async (req, res) => {
 
     res.status(200).send(reportsGroupedByYear.sort((a, b) => b.year - a.year));
   } catch (error) {
-    console.error('Unable to get previous reports %s', error);
+    console.error('Unable to get previous reports %O', error);
```
Comment (abhi-markan) [inline portal-api/src/v1/controllers/utilisation-report-service/previous-reports.controller.js:56]: Same as above.

**Review by abhi-markan — CHANGES_REQUESTED** [review]
Couple of comments to address please.

**Review by abhi-markan — APPROVED** [review]

**Review by ChristianMcc — APPROVED** [review]

**Pull request merged**